kratos/utilities/voxel_utilities.cpp

```diff
@@ -0,0 +1,307 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortes
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "voxel_utilities.h"
+
+namespace Kratos 
+{
+constexpr std::size_t VoxelUtilities::mNeighbours[4][2];
+
+/**********************************************************************************/
+/**********************************************************************************/
+
+double VoxelUtilities::NodesApproximation(const GeometryType& rVoxel) {
+    double volume = 0;
+    const auto& nodes = rVoxel.Points(); //PointsArrayType
+    const std::size_t n = nodes.size();
+    for (std::size_t i = 0; i < n; i++) {
+        if (nodes[i].GetSolutionStepValue(DISTANCE) > 0) {
+            volume+=(1.0/nodes.size()); 
```

Comment:
I would use at least n instead of nodes.size() and for better performance, I would have a const variable for 1./n outside the loop:
```c++
const double nodal_volume_fraction = 1.0 / n;
for(...)
    volume_fraction += nodal_volume_fraction;

```